extensions/EXT/EGL_EXT_explicit_device.txt

```diff
@@ -0,0 +1,101 @@
+Name
+
+    EXT_explicit_device
+
+Name Strings
+
+    EGL_EXT_explicit_device
+
+Contributors
+
+    Adam Jackson
+    Nicolai Haehnle
+    Daniel Stone
+
+Contacts
+
+    Adam Jackson <ajax@redhat.com>
+
+Status
+
+    Complete
+
+Version
+
+    Version 2, 2017-08-08
+
+Number
+
+    EGL Extension #122
+
+Extension Type
+
+    EGL device extension
+
+Dependencies
+
+    Requires EGL_EXT_platform_base and EGL_EXT_device_base.
+
+    EGL_EXT_platform_device trivially interacts with this extension.
+
+    This extension is written against the EGL 1.5 Specification.
+
+Overview
+
+    A system may support multiple devices and multiple window systems. For
+    example, a Wayland environment may drive multiple GPUs and support both
+    X11 and EGL clients. In order to realize this, the implementation must
+    allow the client to specify both device and platform explicitly.
+
+    The EGL_EXT_platform_device extension enables the client to create a
+    display from a device, but does so by overloading the native_display
+    argument to eglGetPlatformDisplay; the user passes the device as the
+    native display, and presumably the EGL picks a sensible default for the
+    window system.
+
+    EGL_EXT_explicit_device differs by passing the device as an attribute
+    to eglGetPlatformDisplay. This allows the client to name both the device
+    and the platform.
```

Comment:
@cubanismo Perhaps:

```c
EGLBoolean eglQueryDisplayDevicesEXT(
        EGLenum platform,
        void *native_display,
        EGLint max_devices,
        EGLDeviceEXT *devices,
        EGLint *num_devices
);
```

Where if `platform` is `EGL_PLATFORM_DEVICE_EXT`, `native_display` must be null, and the returned device list is every device on which `eglGetPlatformDisplay(EGL_PLATFORM_DEVICE_EXT)` might succeed.